cli/lib/utils.ts

```diff
@@ -165,3 +170,43 @@ export function isValidClient(
 ): Client is Client & { client_id: string; name: string } {
   return !!Client.name && !!Client.client_id
 }
+
+/**
+ * Compare two scripts and print the diff between them
+ *
+ * Takes scriptType as a convenience parameter to enable nicer printing, but is
+ * otherwise agnostic as to what it's diffing.
+ */
+export function printScriptDiff<
+  T extends RuleDefinition | DBActionScriptDefinition
+>(diffs: [T, Change[]][], scriptType: 'db scripts' | 'rules' = 'rules') {
+  let changesHeaderPrinted = false
+  const upToDateScripts: T[] = []
+  for (const [ruleDef, changes] of diffs) {
+    if (changes.every((part) => !part.added || part.removed)) {
+      upToDateScripts.push(ruleDef)
+      continue
+    }
+    if (!changesHeaderPrinted) {
+      console.log(`[[ Changed ${scriptType} ]]`)
```

Comment:
This is kind of cute, in that the union you supply as an argument is the full string `db scripts`, but I kind of feel like this is a weird way to do types, and maybe it'd be better to have a uniform type param (as suggested, `ScriptType = 'db' | 'rule'`), and then do the name here with a ternary (or a lookup if we ever have more than two script types)